diff --git a/arch/riscv/kernel/setup.c b/arch/riscv/kernel/setup.c
index db20dc630e7e..64159832e68d 100644
--- a/arch/riscv/kernel/setup.c
+++ b/arch/riscv/kernel/setup.c
@@ -234,7 +234,7 @@ void __init setup_arch(char **cmdline_p)
 	setup_bootmem();
 	paging_init();
 	unflatten_device_tree();
-	swiotlb_init(1);
+	//swiotlb_init(1);
 
 #ifdef CONFIG_SMP
 	setup_smp();
diff --git a/arch/riscv/lib/tishift.S b/arch/riscv/lib/tishift.S
index 69abb1277234..17e28bd1488d 100644
--- a/arch/riscv/lib/tishift.S
+++ b/arch/riscv/lib/tishift.S
@@ -12,31 +12,31 @@
  */
   .globl __lshrti3
 __lshrti3:
-  beqz  a2, .L1
-  li    a5,64
-  sub   a5,a5,a2
-  addi  sp,sp,-16
-  sext.w a4,a5
-  blez  a5, .L2
-  sext.w a2,a2
-  sll   a4,a1,a4
-  srl   a0,a0,a2
-  srl   a1,a1,a2
-  or    a0,a0,a4
-  sd    a1,8(sp)
-  sd    a0,0(sp)
-  ld    a0,0(sp)
-  ld    a1,8(sp)
-  addi  sp,sp,16
+//  beqz  a2, .L1
+//  li    a5,64
+//  sub   a5,a5,a2
+//  addi  sp,sp,-16
+//  sext.w a4,a5
+//  blez  a5, .L2
+//  sext.w a2,a2
+//  sll   a4,a1,a4
+//  srl   a0,a0,a2
+//  srl   a1,a1,a2
+//  or    a0,a0,a4
+//  sd    a1,8(sp)
+//  sd    a0,0(sp)
+//  ld    a0,0(sp)
+//  ld    a1,8(sp)
+//  addi  sp,sp,16
   ret
 .L1:
   ret
 .L2:
-  negw  a4,a4
-  srl   a1,a1,a4
-  sd    a1,0(sp)
-  sd    zero,8(sp)
-  ld    a0,0(sp)
-  ld    a1,8(sp)
-  addi  sp,sp,16
+//  negw  a4,a4
+//  srl   a1,a1,a4
+//  sd    a1,0(sp)
+//  sd    zero,8(sp)
+//  ld    a0,0(sp)
+//  ld    a1,8(sp)
+//  addi  sp,sp,16
   ret
